pulsar/impl_partition_producer.go

```diff
@@ -176,6 +176,8 @@ func (p *partitionProducer) ConnectionClosed() {
 func (p *partitionProducer) reconnectToBroker() {
 	p.log.Info("Reconnecting to broker")
 	backoff := internal.Backoff{}
+	// Delay one second to reconnect
+	time.Sleep(1 * time.Second)
```

Comment:
@tuteng I am not sure this is a right fix. It seems to me that we didn't handle the exception `ServiceNotReady` well. I think we need to add logic to handle `ServiceNotReady` and reconnect.

Reply:
I'm not sure if there is logic for multiple retries after a failure on the go client-side. in my test, I found that after the topic was unload, the topic on the broker side was in the fence state, but the client did not seem to be performing multiple retries, the producer side was blocked.
You may be right, correctly handling this error `ServiceNotReady` and then try again many times should be able to solve this problem.

Reply:
Can we step back and figure out if the client handles `ServiceNotReady` correctly before fixing this by adding time before reconnecting?